qiskit/qpy/binary_io/schedules.py

```diff
@@ -71,6 +76,29 @@ def _loads_symbolic_expr(expr_bytes):
     return expr
 
 
+def _format_legacy_qiskit_pulse_v5(pulse_type, envelope, parameters):
+    # In the transition to Qiskit Terra > 0.22, the representation of library pulses was changed from
+    # complex "amp" to float "amp" and "angle". To reflect this, QPY version was bumped to 6. The
+    # existing library pulses in QPY<=5 are handled here separately to conform with the new
+    # representation. To avoid role assumption for "amp" for custom pulses, only the library pulses
+    # are handled this way.
+
+    # List of pulses in the library in QPY version 5 and below:
+    v5_library_pulses = ["Gaussian", "GaussianSquare", "Drag", "Constant"]
+
+    if pulse_type in v5_library_pulses:
+        if isinstance(
+            parameters["amp"], complex
+        ):  # We know that "amp" is in "parameters" for these pulses.
+            parameters["angle"] = np.angle(parameters["amp"])
+            parameters["amp"] = np.abs(parameters["amp"])
+            _amp, _angle = sym.symbols("amp, angle")
+            envelope = envelope.subs(_amp, _amp * sym.exp(sym.I * _angle))
+        else:
+            parameters["angle"] = 0
+    return envelope
```

Comment:
You need to comment `parameters` is mutated, or return it explicitly for readability.